code/modules/chemistry/tools/ampoules.dm

```diff
@@ -23,12 +23,12 @@
 	if(expended || reagents.total_volume <= 0)
 		boutput(user, SPAN_ALERT("[src] is empty!"))
 		return
-	else if(target == user)
+	if(target == user)
 		boutput(user, SPAN_NOTICE("You crack open and inhale [src]."))
 	else
 		user.visible_message(SPAN_ALERT("[user] attempts to force [target] to inhale [src]!"))
-		SETUP_GENERIC_ACTIONBAR(user, target, 3 SECONDS, /mob/proc/inhale_ampoule, list(src, user), src.icon, src.icon_state, null, \
-			list(INTERRUPT_MOVE, INTERRUPT_ATTACKED, INTERRUPT_STUNNED, INTERRUPT_ACTION))
+	SETUP_GENERIC_ACTIONBAR(user, target, 3 SECONDS, /mob/proc/inhale_ampoule, list(src, user), src.icon, src.icon_state, null, \
+		list(INTERRUPT_MOVE, INTERRUPT_ATTACKED, INTERRUPT_STUNNED, INTERRUPT_ACTION))
```

Comment:
Oops lol

Action bars are just meant for applying to others (like how action bar is used for pills for others, but not for yourself)

Think the fix to maintain current functionality would be to just call `user.inhale_ampoule()` in the else if block, then probably need to add a if (user != src) check in the inhale_ampoule proc for the message

Reply:
good catch ty